mirroring/mirroring.go

```diff
@@ -76,115 +73,18 @@ func GetLogInfo(rekorClient *gclient.Rekor) (*models.LogInfo, error) {
 	return logInfoResp.GetPayload(), nil
 }
 
-func GetLogProof(rekorClient *gclient.Rekor, firstSize, lastSize *int64) (*models.ConsistencyProof, error) {
-	params := tlog.NewGetLogProofParams()
-	if *firstSize > 1 {
-		params.FirstSize = firstSize
-	}
-	params.LastSize = *lastSize
-
-	logProofResp, err := rekorClient.Tlog.GetLogProof(params)
-	if err != nil {
-		return nil, err
-	}
-	return logProofResp.GetPayload(), nil
-}
-
-func VerifySignedTreeHead(sth *util.SignedCheckpoint, pubkey string) error {
-	block, _ := pem.Decode([]byte(pubkey))
+func LoadVerifier(pemPubKey string) (signature.Verifier, error) {
+	block, _ := pem.Decode([]byte(pemPubKey))
 	if block == nil {
-		return errors.New("failed to decode public key of server")
+		return nil, errors.New("failed to decode public key of server")
 	}
 
 	pub, err := x509.ParsePKIXPublicKey(block.Bytes)
 	if err != nil {
-		return err
-	}
-
-	// Initialize verfier and verify
-	verifier, err := signature.LoadVerifier(pub, crypto.SHA256)
-	if err != nil {
-		return err
-	}
-
-	if !sth.Verify(verifier) {
-		return errors.New("signature on tree head did not verify")
-	}
-
-	return nil
-}
-
-func VerifyLogConsistency(rekorClient *gclient.Rekor, oldSize int64, oldRootHash []byte) (*util.SignedCheckpoint, error) {
-	logInfo, err := GetLogInfo(rekorClient)
-	if err != nil {
-		return nil, err
-	}
-
-	logProof, err := GetLogProof(rekorClient, &oldSize, logInfo.TreeSize)
-	if err != nil {
-		return nil, err
-	}
-
-	newRoot, err := hex.DecodeString(*logInfo.RootHash)
-	if err != nil {
-		return nil, err
-	}
-
-	proofs := make([][]byte, len(logProof.Hashes))
-	for i, h := range logProof.Hashes {
-		hash, err := hex.DecodeString(h)
-		if err != nil {
-			return nil, err
-		}
-		proofs[i] = hash
-	}
-
-	err = proof.VerifyConsistency(rfc6962.DefaultHasher, uint64(oldSize), uint64(*logInfo.TreeSize), proofs, oldRootHash, newRoot)
-	if err != nil {
-		return nil, err
-	}
-	sth := util.SignedCheckpoint{}
-	if err := sth.UnmarshalText([]byte(*logInfo.SignedTreeHead)); err != nil {
 		return nil, err
 	}
-	return &sth, nil
-}
-
-func VerifyLogInclusion(rekorClient *gclient.Rekor, entryUUID string) error {
```

Comment:
was this just not used?

Reply:
Yep, this was not used. There's more functions in here that aren't used, but they don't have equivalents in rekor/pkg/verify, so I left them.